cli/azd/cmd/pipeline.go

```diff
@@ -34,6 +34,12 @@ func (pc *pipelineConfigFlags) Bind(local *pflag.FlagSet, global *internal.Globa
 		"origin",
 		"The name of the git remote to configure the pipeline to run on.",
 	)
+	local.StringVar(
+		&pc.PipelineAuthTypeName,
+		"auth-type",
+		"",
+		"The authentication type used between the pipeline provider and Azure for deployment (Only valid for GitHub provider)",
+	)
```

Comment:
Is this supposed to be disjoint from `principal-name` and `principal-role`?

Reply:
When setting up oidc an SP is still created, just the password isn't stored in the GH secret, so the user will need a way to BYOSP - bring your own service principle.

Reply:
Do we already have a spec for the entire user scenario with this new `authType`? I suspect that things are going to be slightly confusing now that `oidc` vs `client-secret` works very differently.

And I suspect that this is likely the forcing function for us to have a separate `auth` command for configuring `auth`, rather than overloading `pipeline config` further.

Reply:
OK. I think I misunderstood it earlier, the SP is created, then applied, which feels inline with the current experience. Would be worth briefly documenting the behaviors if-any stand up as different.